File: 백준/Silver/10773. 제로/제로.cc

#include <iostream>
#include <vector>
#include <stack>
using namespace std;
int main()
{
	ios::sync_with_stdio(false);
	cout.tie(NULL);
	cin.tie(NULL);

	int N;
	cin >> N;
	
	stack<int> myStack;

	for (int i = 0; i < N; i++) {
		int temp;
		cin >> temp;
		myStack.push(temp);
		if (myStack.top() == 0) {
			myStack.pop();
			myStack.pop();
		}
		
	}

	long long sum = 0;
	int size = myStack.size();
	for (int i = 0; i < size; i++) {
		sum += myStack.top();
		myStack.pop();
	}

	cout << sum << "\n";
}

File: 백준/Silver/10773. 제로/제로.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {

	
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		Stack<Integer> stack = new Stack<>();
			
		int N = Integer.parseInt(br.readLine());
	
		
		for(int i = 0;i < N;i++) {
			int temp = Integer.parseInt(br.readLine());
			
			stack.push(temp);
			
			if(temp == 0) {
				stack.pop();
				stack.pop();
			}
		}
		
		long sum = 0;
		
		int size = stack.size();
		
		for(int i = 0;i < size;i++) {
			sum += stack.peek();
			stack.pop();
		}
		
		System.out.println(sum);
		
	}

}